### ISSUE TITLE
NVDA does not announce contents of div with role="alert" containing a list

### ISSUE THREAD

**Issue by grabowsd** [nvaccess/nvda#14990]
### Steps to reproduce:
https://jsfiddle.net/gdaria/9kcy7fo3/15/

This is similar issue to https://github.com/nvaccess/nvda/issues/8557
When a list `<ul>` and heading `<h4>` are wrapped in a `div` with `role=alert` the list and the heading will not be read by NVDA.

```
  <div id="errorList" class="hidden" role="alert">
    <h4>
      Error List
    </h4>
    <ul>
      <li>Error 1</li>
      <li>Error 2</li>
      <li>Error 3</li>
    </ul>
  </div>
```

When a list is removed from the div the heading will be announced or when `role=alert` is added to `<ul>` the list will be announced.   

```
    <ul id="errorList" class="hidden" role="alert">
      <li>Error 1</li>
      <li>Error 2</li>
      <li>Error 3</li>
    </ul>
```

### Actual behavior:

When button is clicked NVDA only says "Alert" and it doesn't read a heading or a list. 

### Expected behavior:

When button is clicked NVDA should announce full alert message (heading + list)

#### NVDA version:
NVDA Version 2023..1.0.27913

#### Windows version:
Windows 10 Enterprise Version 10.0.19045 Build 19045

#### Name and version of other software in use when reproducing the issue:
Microsoft Edge: Version 113.0.1774.57 (Official build) (64-bit)
Chrome Version 114.0.5735.110 (Official Build) (64-bit)
 Firefox 113.0.2 (64-bit)

**Comment by alexarnaud** [comment]
Hello,

I confirm, I reproduce the issue with NVDA and not with JAWS.

Thanks a lot for reporting it.

**Comment by Adriani90** [comment]
Comparing the two code snippets above, testing with NvDA 2024.4 Beta:
* In Firefox both code samples do not trigger any allert message
* In Chromium, the first one triggers only the allert role being announced, while the second works as expected where NVDA reports both the role and the complete message.

cc: @jcsteh since it affects Firefox mostly.

**Comment by PtitBen** [comment]
Hello, 

This issue seems to have been triaged and has been given a priority level a little bit over a year ago. I was wondering if this is still on the books to be fixed? 

We are experiencing the same issue where the heading (first child) of our alert copmonent is skipped by NVDA. Our only real option if this bug wasn't addressed for NVDA would be to slightly degrade the experience on other screen readers/clients to try to get the full content to be read by NVDA. 

Thank you in advance for a potential update!

**Comment by alexarnaud** [comment]
Hello,

It remains this technic in the meantime:
[Make a screen reader talk with JavaScript and ARIA](https://a11y-guidelines.orange.com/en/articles/make-a-screen-reader-talk/)

**Comment by Serhansolo** [comment]
Hi guys, I stumbled upon this issue. Do you have an ETA for when this will be fixed? I also have to resort to the measures @PtitBen is stating.  I remember I could put a simple heading with some text and lists to be read.

**Comment by seanbudd** [comment]
Hi @Serhansolo - unfortunately we are unable to provide timeframes for fixes. The majority of bug fixes in NVDA are performed by volunteers from the community. This issue is triaged, meaning any community member can open a PR to address this.
